shakytweaks/src/main/java/com/mindsea/shakytweaks/ShakyTweaks.kt

```diff
@@ -86,6 +102,28 @@ object ShakyTweaks {
         override fun tweakValueResolver(): TweakValueResolver = tweakValueResolver
     }
 
+    private class LifecycleObserver : LifecycleEventObserver {
+        override fun onStateChanged(source: LifecycleOwner, event: Lifecycle.Event) {
+            when (event) {
+                Lifecycle.Event.ON_RESUME -> {
+                    sensorManager.registerListener(
+                        shakeDetector,
+                        accelerometer,
+                        SensorManager.SENSOR_DELAY_UI
+                    )
+                }
+                Lifecycle.Event.ON_STOP -> {
+                    sensorManager.unregisterListener(shakeDetector)
+                }
+                Lifecycle.Event.ON_DESTROY -> {
```

Comment:
Still having problems in Stitch app.
In that scenario, there is a Splash screen and then the MainActivity. From what I saw, when you finish one activity A and start activity B, it happens that ON RESUME of B is called before ON_DESTROY of A. So when A is destroyed, there is no more active listeners 🤯 

I think the solution here is to keep hold of a `sensorManager` for each `lifecycle` registered. That might be tricky.

Another option, maybe leave the `sensorManager` always registered (as it was before this PR), but on line 66, before starting the tweaks activity, check if `lifecycle` is at least on resumed state. (not sure if this would work or have other problems though)

Reply:
I created a new branch, just to generate a new snaphsot. You can use:
`def shaky_tweaks_version = 'debug-test-SNAPSHOT'`
To test this version

Reply:
I was reading this [link](https://developer.android.com/guide/components/activities/activity-lifecycle.html#soafa). 
Maybe our solution it would be we change `onStop` for `onPause`. 
In your scenario, the `SplashActivity` starts another. So, the last state before creating the new activity that `SplashActivity` travels is `onPause` (according to the documentation, I'm testing this condition).

I sent an update. I'm testing here as well. I hope it works.

Reply:
I tested here, my attempt is not working. I think we need to do what you said, but I don't know how :cry: